esx_service/vmdk_ops.py

```diff
@@ -402,8 +402,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
 # Return error, or None for OK
 def removeVMDK(vmdk_path):
     logging.info("*** removeVMDK: %s", vmdk_path)
-    cmd = "{0} {1}".format(VMDK_DELETE_CMD, vmdk_path)
 
+    # Check the current volume status
+    attached, uuid, attach_as = getStatusAttached(vmdk_path)
+    if attached:
```

Comment:
@pdhamdhere 
Using the api has the same destructive effect as vmkfstools -U.

Using vmkfstools -D to find out if its in use works, but not for the volume.vmdk, In my case was the volume-flat.vmdk.

According with [this document](https://kb.vmware.com/selfservice/microsites/search.do?language=en_US&cmd=displayKC&externalId=10051) we should search for the owner being set.

Do we have identified the conditions or code paths for the stale metadata cases? If not a async mechanism (in relation to the ops), we could have consistency/integrity checks at the end of each op that change metadata (while we still hold the data for saving or update).

vmkfstools -D /vmfs/volumes/datastore2/dockvols/vol1174812.vmdk
```
Lock [type 10c00001 offset 99715072 v 3122, hb offset 3170304
gen 169, mode 0, owner 00000000-00000000-0000-000000000000 mtime 139245
num 0 gblnum 0 gblgen 0 gblbrk 0]
Addr <4, 222, 33>, gen 3104, links 1, type reg, flags 0, uid 0, gid 0, mode 600
len 566, nb 0 tbz 0, cow 0, newSinceEpoch 0, zla 4305, bs 8192
```

vmkfstools -D /vmfs/volumes/datastore2/dockvols/vol1174812-671ef9f932576c16.vmfd
```
Lock [type 10c00001 offset 99717120 v 3132, hb offset 3170304
gen 169, mode 0, owner 00000000-00000000-0000-000000000000 mtime 139221
num 0 gblnum 0 gblgen 0 gblbrk 0]
Addr <4, 222, 34>, gen 3108, links 1, type reg, flags 0, uid 0, gid 0, mode 600
len 4096, nb 1 tbz 0, cow 0, newSinceEpoch 1, zla 2, bs 8192

```
***Field owner is set***
vmkfstools -D /vmfs/volumes/datastore2/dockvols/vol1174812-flat.vmdk
```
Lock [type 10c00001 offset 99713024 v 3114, hb offset 3170304
gen 169, mode 1, owner 581272ea-1208c2da-516b-b8aeedeb6b16 mtime 139203
num 0 gblnum 0 gblgen 0 gblbrk 0]
Addr <4, 222, 32>, gen 3103, links 1, type reg, flags 0, uid 0, gid 0, mode 600
len 104857600, nb 15 tbz 0, cow 0, newSinceEpoch 15, zla 1, bs 1048576
```